internal/daemon/cluster/handlers/worker_service.go

```diff
@@ -82,16 +77,20 @@ func (ws *workerServiceServer) Status(ctx context.Context, req *pbs.StatusReques
 		}
 	}
 
-	wConf := servers.NewWorkerStatus(reqServer.PrivateId,
-		// TODO: Change the name to its own field to make this explicit.
+	wConf := servers.NewWorkerForStatus(scope.Global.String(),
 		servers.WithName(reqServer.PrivateId),
 		servers.WithAddress(reqServer.Address),
 		servers.WithWorkerTags(workerTags...))
-	controllers, _, err := serverRepo.UpsertWorkerStatus(ctx, wConf, servers.WithUpdateTags(req.GetUpdateTags()))
+	wrk, err := serverRepo.UpsertWorkerStatus(ctx, wConf, servers.WithUpdateTags(req.GetUpdateTags()))
 	if err != nil {
 		event.WriteError(ctx, op, err, event.WithInfoMsg("error storing worker status"))
 		return &pbs.StatusResponse{}, status.Errorf(codes.Internal, "Error storing worker status: %v", err)
 	}
+	controllers, err := serverRepo.ListControllers(ctx)
```

Comment:
just a general note about the tests: it might be nice to add some failure conditions unit tests.  I know it's not typical for our services tests, but perhaps we can start making it more normal?